src/ray/raylet/node_manager.cc

```diff
@@ -949,8 +951,14 @@ void NodeManager::ProcessRegisterClientRequestMessage(
   auto message = flatbuffers::GetRoot<protocol::RegisterClientRequest>(message_data);
   Language language = static_cast<Language>(message->language());
   WorkerID worker_id = from_flatbuf<WorkerID>(*message->worker_id());
-  auto worker = std::make_shared<Worker>(worker_id, message->worker_pid(), language,
-                                         message->port(), client, client_call_manager_);
+  WorkerProcessHandle proc =
+      worker_pool_.FindStartingWorkerByProcessId(language, message->worker_pid());
+  if (!proc) {
+    RAY_LOG(WARNING) << "Unrecognized worker";
```

Comment:
Thanks, could you also add some comments on why this could happen?

Reply:
andn it would be nice to also print the `worker_id` as well as `pid` in this case

Reply:
Actually it seems this is already happening and the build is broken :( any ideas?

Reply:
I think it's because the driver is not a child of raylet, but it still needs to register.

Basically if `message->is_worker()` is false, the registration request should still be processed.

Reply:
Ohh I see! Okay let me try that, thanks!

Reply:
It seems to be possibly working, but I'm not sure if the build breakage is due to a different issue or not. It might be just due to the flakiness?